File: src/main/java/edu/austral/starship/base/framework/GameFramework.java
package edu.austral.starship.base.framework;

import processing.core.PGraphics;
import processing.event.KeyEvent;

import java.util.Set;

public interface GameFramework {
    void setup(WindowSettings windowsSettings, ImageLoader imageLoader);

    void draw(PGraphics graphics, float timeSinceLastDraw, Set<Integer> keySet);

    void keyPressed(KeyEvent event);

    void keyReleased(KeyEvent event);
}

File: src/main/scala/edu/austral/starship/scala/base/framework/GameFramework.scala
package edu.austral.starship.scala.base.framework

import processing.core.PGraphics
import processing.event.KeyEvent

trait GameFramework {
  def setup(windowsSettings: WindowSettings, imageLoader: ImageLoader): Unit

  def draw(graphics: PGraphics, timeSinceLastDraw: Float, keySet: Set[Int]): Unit

  def keyPressed(event: KeyEvent): Unit

  def keyReleased(event: KeyEvent): Unit
}
